File: AzW.Web.Portal/src/models/VM.js
import ResourceType from './ResourceType'
export default class VM
{
    constructor (){
        this.GraphModel = {
            Id: '',
            ResourceType: ResourceType.VM(),
            DisplayName: ''
        };

        this.ProvisionContext = {
            ResourceType: ResourceType.VM(),

            Name: 'vm-webserver',
            Location: 'westus',

            HasPublicIP: false,
            PublicIPName: '',

            VNetName: '',
            SubnetName: '',

            AdminUsername: 'AzureUser',
            AdminPassword: '',

            VMPublisher: '',
            VMOffer: '',
            VMSKU: '',
            VMVersion: '',
            VMPublisherSearchableName: '',

            SizeName: 'STANDARD_DS2_V2',
            IsLinux: false,
            SSHPublicKey: ''

        };
        this.CalculatorContext = {
            IsFree: false,
            Tier: ''
        }
    }
}

File: Portal.Services/AzW.Model/Provision/VM.cs

using System.Collections.Generic;

namespace AzW.Model
 {
     public class VM : Resource
     {
          public string Location { get; set; }
          public string ResourceGroupName { get; set; }
          public string AddressSpace { get; set; }
          public string VNetName { get; set; }
          public string SubnetName { get; set; }
          public bool HasPublicIP { get; set; }
          public string PublicIPName { get; set; }

          public string VMPublisher { get; set; }
          public string VMOffer { get; set; }
          public string VMSKU { get; set; }
          public string VMVersion { get; set; }
          public string AdminUsername { get; set; }
          public string AdminPassword { get; set; }
          public string SizeName { get; set; }
          public bool IsLinux { get; set; }
     }
 }